diff --git a/CMakeLists.txt b/CMakeLists.txt
index e0b73e37..e8c5592a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -200,6 +200,13 @@ if(HAVE_ATOMIC)
     target_link_libraries(sysrepo atomic)
 endif()

+# find required libraries
+find_library(ZLOG zlog)
+if(NOT ZLOG)
+    message(FATAL_ERROR "zlog must be installed.")
+endif()
+target_link_libraries(sysrepo ${ZLOG})
+
 # libyang, check version
 find_package(YANG REQUIRED)
 target_link_libraries(sysrepo ${LIBYANG_LIBRARIES})
diff --git a/src/executables/sysrepo-plugind.c b/src/executables/sysrepo-plugind.c
index ca22140e..270c45e1 100644
--- a/src/executables/sysrepo-plugind.c
+++ b/src/executables/sysrepo-plugind.c
@@ -38,6 +38,8 @@
 #include <fcntl.h>
 #include <dlfcn.h>

+#include <zlog.h>
+
 #include "sysrepo.h"
 #include "bin_common.h"

@@ -390,6 +392,9 @@ main(int argc, char** argv)
         goto cleanup;
     }

+    /* set logging */
+    dzlog_init("/opt/etc/zlog.conf", "sysrepo-plugind");
+
     /* load plugins */
     if (load_plugins(&plugins, &plugin_count)) {
         goto cleanup;
diff --git a/src/executables/sysrepocfg.c b/src/executables/sysrepocfg.c
index 2172b869..07ed5b7a 100644
--- a/src/executables/sysrepocfg.c
+++ b/src/executables/sysrepocfg.c
@@ -36,6 +36,8 @@

 #include <libyang/libyang.h>

+#include <zlog.h>
+
 #include "sysrepo.h"
 #include "bin_common.h"

@@ -776,6 +778,7 @@ main(int argc, char** argv)
     }

     /* set logging */
+    dzlog_init("/opt/etc/zlog.conf", "sysrepocfg");
     sr_log_stderr(log_level);

     /* create connection */
diff --git a/src/executables/sysrepoctl.c b/src/executables/sysrepoctl.c
index f9952b8e..39859c86 100644
--- a/src/executables/sysrepoctl.c
+++ b/src/executables/sysrepoctl.c
@@ -31,6 +31,8 @@

 #include <libyang/libyang.h>

+#include <zlog.h>
+
 #include "sysrepo.h"
 #include "bin_common.h"

@@ -600,6 +602,7 @@ main(int argc, char** argv)
     }

     /* set logging */
+    dzlog_init("/opt/etc/zlog.conf", "sysrepoctl");
     sr_log_stderr(log_level);

     /* set callback */
diff --git a/src/log.c b/src/log.c
index 1be87ce0..07883e85 100644
--- a/src/log.c
+++ b/src/log.c
@@ -27,6 +27,8 @@
 #include <assert.h>
 #include <syslog.h>

+#include <zlog.h>
+
 sr_log_level_t stderr_ll = SR_LL_NONE;  /**< stderr log level */
 sr_log_level_t syslog_ll = SR_LL_NONE;  /**< syslog log level */
 int syslog_open;                        /**< Whether syslog was opened */
